account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -745,6 +754,7 @@ def create(self, mvals):
         models = self.env["account.reconcile.model"].search(
             [
                 ("rule_type", "in", ["invoice_matching", "writeoff_suggestion"]),
+                ("company_id", "in", result.mapped("company_id").ids),
```

Comment:
With relation fields, no need to put `mapped`:

```suggestion
                ("company_id", "in", result.company_id.ids),
```